dask/array/core.py

```diff
@@ -989,6 +989,13 @@ def finalize(results):
     return unpack_singleton(results)
 
 
+def finalize(results):
+    res = _finalize(results)
+    if type(res) == np.ndarray and not res.flags.writeable:
```

Comment:
I do not use `isinstance()` or otherwise it triggers on masked-arrays.

Reply:
Maybe the suggestion below addresses the masked array case?

Reply:
I did not remove the `type(...)` to avoid what I think is unnecessary, but let me know.

Reply:
Yeah that's fine. We can always revisit if needed.